lib/nvme/nvme_pcie.c

```diff
@@ -1841,7 +1841,24 @@ nvme_pcie_qpair_check_timeout(struct spdk_nvme_qpair *qpair)
 		 * Request has timed out. This could be i/o or admin request.
 		 * Call the registered timeout function for user to take action.
 		 */
-		ctrlr->timeout_cb_fn(ctrlr->timeout_cb_arg, ctrlr, qpair, tr->cid);
+
+		if (tr->req->cmd.opc == SPDK_NVME_OPC_ABORT) {
+			/*
+			 * An abort command has timed out. Modify timeout_count value to be
+			 * greater than 1 to cause a controller reset from timeout callback
+			 * function.
+			 */
+			tr->req->timeout_count = 2;
+		} else {
+			tr->req->timeout_count++;
+		}
+
+		tr->submit_tick = spdk_get_ticks();
+		ctrlr->timeout_cb_fn(ctrlr,
+				     qpair,
+				     ctrlr->timeout_cb_arg,
+				     tr->req->timeout_count,
+				     tr->req->cmd.cid);
```

Comment:
Jim, the code in nvme_pcie_qpair_process_completions(..) function checks to see if the controller state is NVME_CTRLR_STATE_READY before it calls the timeout_cb_fn function.